### PR TITLE
mpv: make package field nullable

### PR DESCRIPTION
**Pull request by smoren-brk**
if package is provided by the system, we still want to be able to configure the package via home-manager by setting package to null

### Description

Make mpv package field nullable so we can skip installation if we already provide package in system config

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `nix fmt` or
      `nix-shell -A dev --run treefmt`.

- [x] Code tested through `nix build .#test-all`
      or a targeted `nix run .#tests -- <pattern>`.

- [x] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

  ```
  {component}: {description}

  {long description}
  ```

  See [CONTRIBUTING](https://nix-community.github.io/home-manager/#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [ ] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/a51598236f23c89e59ee77eb8e0614358b0e896c/modules/programs/lesspipe.nix#L11).
  - [ ] Generate a news entry. See [News](https://nix-community.github.io/home-manager/index.xhtml#sec-news)
  - [ ] Basic tests added. See [Tests](https://nix-community.github.io/home-manager/index.xhtml#sec-tests)

- If this PR adds an exciting new feature or contains a breaking change.
  - [ ] Generate a news entry. See [News](https://nix-community.github.io/home-manager/index.xhtml#sec-news)

### CODE REVIEW

File: modules/programs/mpv.nix [hunk 1]
```diff
@@ -68,7 +68,9 @@ let
   ];
 
   mpvPackage =
-    if wrapperRequiresOverride then
+    if cfg.package == null then
+      null
+    else if wrapperRequiresOverride then
```
Comment (ambroisie) [inline modules/programs/mpv.nix:73]: Add an assertion that `cfg.package != null` if we require override.

Note: according to NixOS/nixpkgs#484501, `extraMakeWrapperArgs` is unnecessary, perhaps we should clean it up (different PR).

**Review by ambroisie — CHANGES_REQUESTED** [review]
Please add a test to check the "package is null but we require an override" scenario.

**Comment by smoren-brk** [comment]
Thank you for taking a look at this !
